src/streams/stransport.c

```diff
@@ -49,6 +51,67 @@ typedef struct {
 	git_cert_x509 cert_info;
 } stransport_stream;
 
+static int stransport_setup_cipher_list(SSLContextRef ctx)
+{
+	SSLCipherSuite *supported_ciphers = malloc(sizeof(*supported_ciphers));
+	SSLCipherSuite *enabled_ciphers = NULL;
+	size_t num_supported = 0, num_enabled = 0;
+	OSStatus ret;
+	const char *cipher_list = git_libgit2__ssl_ciphers();
+	const char *name;
+	size_t len;
+	int error = 0;
+
+	ret = SSLGetNumberSupportedCiphers(ctx, &num_supported);
+	if (ret != noErr) {
+		return stransport_error(ret);
+	}
+
+	supported_ciphers = git__calloc(num_supported, sizeof(*supported_ciphers));
+	enabled_ciphers = git__calloc(num_supported, sizeof(*enabled_ciphers));
+
+	ret = SSLGetSupportedCiphers(ctx, supported_ciphers, &num_supported);
+	if (ret != noErr) {
+		stransport_error(ret);
+		error = -1;
+		goto cleanup;
+	}
+
+	while (git_tls_enabled_ciphers(&name, &len, &cipher_list) == 0) {
+		const git_tls_cipher *cipher;
+		size_t idx;
+
+		if (git_tls_cipher_lookup(&cipher, name, len) < 0) {
+			/* TODO: cipher not found, warn ? */
```

Comment:
Warning wouldn't make a lot of sense in case where this is going to be our generic interface for all implementations. Our set of enabled-by-default ciphers shouldn't be what is implemented by all the TLS implementations we support, but instead by what's secure and available in at least a subset of them. Thus warnings wouldn't make sense in that case, as it is expected that some of our preferred ciphers are not available in some of our TLS backends.